packages/mui-system/src/colorManipulator/colorManipulator.js

```diff
@@ -33,6 +33,14 @@ export function hexToRgb(color) {
     colors = colors.map((n) => n + n);
   }
 
+  if (process.env.NODE_ENV !== 'production') {
+    if (colors.length >= 4 && Number.isNaN(parseInt(colors[3], 16))) {
```

Comment:
Would something like this work? It would also catch spaces at the start.

```suggestion
    if (colors.length >= 4 && colors[3].length !== color.length) {
```

Reply:
Your suggestion will be false alert for `color` = 'ff5252' and `colors[3]` = 1.
I don't think `colors[3]` and `color` should be compared because `color` is a hex, but `colors` is rgba channels.

Reply:
Sorry, this was the suggestion I had in mind:

```suggestion
    if (color.length !== color.trim().length) {
```